File: C#HandsOn/Ex11/Employee.cs
using System;

namespace Ex11
{

    public class Employee
    {
        //性
        private string lastname;
        //名
        private string firstname;
        //年齢
        private int age;

        //11-8
        static public string CompanyName = "ビプ商事";

        public static void ClassInfo()
        {
            Console.WriteLine("Employeeクラスです");
        }
        
        //プロパティ
        public string Lastname
        {
            get
            {
                return lastname;
            }
            set
            {
                if(!string.IsNullOrEmpty(value))
                {
                    lastname = value;
                }
            }
        }
        public string Firstname
        {
            get
            {
                return firstname;
            }
        }
        
        /*----------------------------------*/
        //コンストラクタ
         //引数なし
        public Employee()
        {
            lastname = "姓なし";
            firstname = "名なし";
            age = 20;
        }
         //引数あり
        public Employee(string lname,string fname,int n)
        {
            lastname = lname;
            firstname = fname;
            age = n;
        }   
        //メソッド
        public string GetFullName()
        {
            return lastname + firstname;
        }
        public string GetFullName(string delim)
        {
            return lastname + delim + firstname;
        }
        public virtual void PrintInfo()
        {
            Console.WriteLine("姓名：" + lastname + firstname);
            Console.WriteLine("年齢" + age);
        }


    }
} 

File: VBHandsOn/Ex11/Employee.vb
Public Class Employee
    'フィールド定義
    Private lastnameValue As String
    Private firstnameValue As String
    Private age As Integer
    Public Shared companyName As String = "ビプ商事"

    'プロパティ定義
    Public Property Lastname() As String
        Get
            Return lastnameValue
        End Get
        Set(value As String)
            If Not String.IsNullOrEmpty(value) Then
                lastnameValue = value
            End If
        End Set
    End Property
    Public ReadOnly Property Firstname() As String
        Get
            Return firstnameValue
        End Get
    End Property

    'コンストラクター定義
    Public Sub New()
        lastnameValue = "姓なし"
        firstnameValue = "名なし"
        age = 20
    End Sub
    Public Sub New(lname As String, fname As String, n As Integer)
        lastnameValue = lname
        firstnameValue = fname
        age = n
    End Sub

    'メソッド定義
    Public Function GetFullName() As String
        Return Lastname & Firstname
    End Function
    Public Function GetFullName(delim As String) As String
        Return Lastname & delim & Firstname
    End Function
    Public Overridable Sub PrintInfo()
        Console.WriteLine("姓名：" & Lastname & Firstname)
        Console.WriteLine("年齢：" & age)
    End Sub

    Shared Sub ClassInfo()
        Console.WriteLine("Employeeクラスです")
    End Sub
End Class
